src/dom.js

```diff
@@ -530,6 +530,40 @@ export function getDataParamsFromAttributes(element, opt_computeParamNameFunc,
 }
 
 /**
+ * Returns link element url components as url parameters key-value pairs.
+ * e.g. <a href="https://example.com/index.html?query#hash"
+ * -> {
+ *   clickHostname: example.com,
+ *   clickProtocol: https,
+ *   clickProtocol: /index.html,
+ *   clickQuery: query,
+ *   clickHash: hash,
+ *   clickUrl: https://example.com/index.html?query#hash
+ * }
+ * @param {!Element} element
+ * @param {function(string):string=} opt_computeParamNameFunc to compute the parameters
+ *    name, get passed the camel-case parameter name.
+ * @return {!JsonObject}
+ */
+export function getDataParamsFromLinkUrl(element, opt_computeParamNameFunc) {
+  if (element.tagName.toUpperCase() != 'A') {
+    return dict();
+  }
+  const computeParamNameFunc = opt_computeParamNameFunc || (key => key);
+  const params = dict();
+  params[computeParamNameFunc('clickHostname')] = element.hostname;
+  params[computeParamNameFunc('clickProtocol')]
+    = element.protocol.replace(/:$/, '');
+  params[computeParamNameFunc('clickPathname')] = element.pathname;
+  params[computeParamNameFunc('clickQuery')] = element.search.slice(1);
+  params[computeParamNameFunc('clickHash')] = element.hash.slice(1);
+  params[computeParamNameFunc('clickUrl')]
```

Comment:
why not `element.href` here?

Reply:
It is my misunderstanding. I thought that ```element.href``` would return the value of the attribute(return relative path if it is a relative path).

But in here, I use the ```url.js``` library.